src/dialog.jsx

```diff
@@ -379,27 +375,30 @@ let Dialog = React.createClass({
     }
   },
 
-  _onShow() {
-    if (this.props.onShow) this.props.onShow();
+  _show() {
```

Comment:
I think that this method is only called once. Can you move it where it's called?